[8.0][IMP][base_export_manager] Improve UX

yajo opened the pull request:
- Improve user instructions in README.
- Require some required fields.
- Allow to select models from a list.
- Allow to select up to 3 fields from dynamic lists.
- Improve translations.
- More tests.
- Translate column labels.

Some methods have been renamed, so version tag is raised to 8.0.2.0.0.

yajo committed:
Improve UX for base_export_manager.

- Improve user instructions in README.
- Require some required fields.
- Allow to select models from a list.
- Allow to select up to 3 fields from dynamic lists.
- Improve translations.
- More tests.
- Translate column labels.

Some methods have been renamed, so version tag is raised to 8.0.2.0.0.

elicoidal commented on base_export_manager/README.rst:
```diff
@@ -43,6 +67,8 @@ Contributors
+* Jairo Llopis <yajo.sk8@gmail.com>
 
 Maintainer
 ----------
```
https

yajo replied:
Fixed.

elicoidal commented on base_export_manager/__openerp__.py:
```diff
@@ -23,11 +23,12 @@
 {
```
Please take the opportunity to simplify headers.

yajo replied:
Fixed.

rafaelbn commented:
Travis fails

yajo committed:
Fix test.

yajo commented on base_export_manager/models/ir_exports.py:
```diff
@@ -0,0 +1,65 @@
+    def _inverse_model_id(self):
+        """Get the resource from the model."""
+        for s in self:
+            s.resource = s.model_id.model
```
You can see this method is not being called in the runbot when you select a `model_id`. I'd thank help on why.

yajo commented on base_export_manager/models/ir_exports_line.py:
```diff
@@ -30,28 +11,119 @@ class IrExportsLine(models.Model):
+    model1_id = fields.Many2one(
+        string="First model",
+        readonly=True,
+        related="export_id.model_id")
```
This field is not working in runbot. It is always false, no matter the value of `export_id.model_id`. I'd thank help on that too.

rafaelbn commented:
Maybe @JavierIniesta could help

yajo committed:
Make inverse method be called at view time.

rafaelbn commented:
Rebuilding for testing

cc @antespi @pedrobaeza

moylop260 commented:
👍 
Please, Could you check the rst syntax error of readme https://travis-ci.org/OCA/server-tools/jobs/99268857?

rafaelbn commented:
@Yajo  Tested. I'm not able to create or edit an export profile. Could you please check it?

![2016-05-28_17-14-17](https://cloud.githubusercontent.com/assets/8736623/15628070/e0ebd9b4-24f7-11e6-83e5-ec802b439bae.png)

![2016-05-28_17-15-21](https://cloud.githubusercontent.com/assets/8736623/15628072/ea806e90-24f7-11e6-854a-af1ce419e673.png)

yajo commented:
@Tecnativa

rafaelbn commented:
Please @Yajo review this PR, thanks!

yajo committed:
README & author updates.

yajo committed:
Hardcode constraints in the `create` method instead of using normal ones.

Depending on the context where the record is created, it gets `resource` or `model_id`. The problem is that Odoo checks constrains before inverses, so constrains would fail always.

Test added to ensure future versions contemplate all use cases.

yajo committed:
Allow to reorder fields.

yajo commented:
All should be working not, including ability to reorder fields.

rafaelbn commented:
Thank-you, perfect! :+1: :smile: :rocket: tested in runbot! After 7 months waiting for this...

Please @moylop260 could you review here again?

cc @lasley done! :wink:

moylop260 commented:
I'm agree yet

pedrobaeza commented:
Tried on runbot :+1:

rafaelbn commented:
@Yajo travis is failing, please take a look. When travis will be green this PR is ready to be merged

rafaelbn commented:
@Yajo travis fails again

yajo commented:
It seems [the guilty module is fetchmail_attach_from_folder](https://travis-ci.org/OCA/server-tools/jobs/147705061#L1765-L1771).

pedrobaeza commented:
Please fix it (here or in other PR and rebase this one)

lasley commented:
Code + functional 👍  

Can confirm, Travis errors are all unrelated and were on 8.0 a few iterations ago. Seems if we rebase on top of 8.0 now though, we will have a [Lint error](https://travis-ci.org/OCA/server-tools/builds/147159732) to replace it 😆

pedrobaeza commented:
OK, I'm merging this now then and I'll commit directly the fix for the linter.
